# Pertemuan 2



In [2]:
# Import library
import pandas as pd
import numpy as np

In [3]:
# Membaca dataset Titanic

link_data = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"

titanic = pd.read_csv(link_data)

# Menampilkan 5 data pertama
titanic.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


Dataset Titanic berisi informasi penumpang kapal Titanic seperti nama, usia, jenis kelamin, kelas penumpang, hingga status keselamatan.

In [5]:
# Q1 - Berapa jumlah baris dan kolom pada dataset Titanic? Kolom apa saja yang tersedia?

print("Ukuran dataset :", titanic.shape)

print("\nDaftar kolom:")
print(titanic.columns.tolist())

Ukuran dataset : (891, 12)

Daftar kolom:
['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']


Dataset memiliki 891 baris dan 12 kolom.  
Setiap kolom menyimpan informasi berbeda mengenai penumpang Titanic.

In [6]:
# Q2 - Kolom mana saja yang memiliki missing values? Berapa jumlah dan proporsinya (%) per kolom?

jumlah_kosong = titanic.isna().sum()

persentase_kosong = (jumlah_kosong / len(titanic)) * 100

ringkasan_missing = pd.DataFrame({
    "Jumlah Data Kosong": jumlah_kosong,
    "Persentase (%)": persentase_kosong
})

# Menampilkan hanya kolom yang memiliki missing value
ringkasan_missing[ringkasan_missing["Jumlah Data Kosong"] > 0]

,Jumlah Data Kosong,Persentase (%)
Age,177,19.865320
Cabin,687,77.104377
Embarked,2,0.224467


Beberapa kolom masih memiliki data kosong, terutama pada kolom Cabin dan Age.

In [7]:
# Q3 - Berapa persen penumpang yang selamat (Survived=1)? Gunakan operasi NumPy atau Pandas

total_selamat = titanic["Survived"].sum()

jumlah_penumpang = len(titanic)

persen_selamat = (total_selamat / jumlah_penumpang) * 100

print(f"Jumlah penumpang selamat : {total_selamat}")
print(f"Persentase keselamatan : {persen_selamat:.2f}%")

Jumlah penumpang selamat : 342
Persentase keselamatan : 38.38%


Hasil perhitungan menunjukkan bahwa kurang dari setengah penumpang Titanic berhasil selamat.

In [8]:
# Q4 - Filter penumpang wanita dari kelas 1 (Pclass=1). Berapa jumlahnya dan berapa ratarata usianya?

female_first_class = titanic[
    (titanic["Sex"] == "female") &
    (titanic["Pclass"] == 1)
]

print("Total penumpang perempuan kelas 1 :", len(female_first_class))

print("Rata-rata usia :", round(female_first_class["Age"].mean(), 2), "tahun")

Total penumpang perempuan kelas 1 : 94
Rata-rata usia : 34.61 tahun


Penumpang perempuan pada kelas 1 cenderung memiliki usia dewasa.

In [9]:
# Q5 - Gunakan groupby untuk menghitung tingkat keselamatan (mean Survived) per kelas penumpang. Kelas mana yang paling tinggi?

keselamatan_kelas = titanic.groupby("Pclass")["Survived"].mean() * 100

print(keselamatan_kelas)

kelas_teraman = keselamatan_kelas.idxmax()

print(f"\nKelas dengan tingkat keselamatan tertinggi adalah kelas {kelas_teraman}")

Pclass
1    62.962963
2    47.282609
3    24.236253
Name: Survived, dtype: float64

Kelas dengan tingkat keselamatan tertinggi adalah kelas 1


Penumpang kelas 1 memiliki peluang bertahan hidup paling tinggi dibanding kelas lainnya.

In [10]:
# Q6 - Buat analisis tambahan berdasarkan variabel lain yang Anda anggap menarik. Sertakan interpretasi singkat.

keselamatan_gender = titanic.groupby("Sex")["Survived"].mean() * 100

print(keselamatan_gender)

print("\nAnalisis tambahan berdasarkan usia")

titanic["Kategori_Umur"] = titanic["Age"].apply(
    lambda x: "Anak-anak" if x < 18 else "Dewasa"
)

hasil_umur = titanic.groupby("Kategori_Umur")["Survived"].mean() * 100

print(hasil_umur)

Sex
female    74.203822
male      18.890815
Name: Survived, dtype: float64

Analisis tambahan berdasarkan usia
Kategori_Umur
Anak-anak    53.982301
Dewasa       36.118252
Name: Survived, dtype: float64


Dari hasil analisis terlihat bahwa perempuan dan anak-anak memiliki tingkat keselamatan lebih tinggi.  
Hal ini berkaitan dengan prioritas evakuasi pada saat tragedi Titanic terjadi.